hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSImageFormatProtobuf.java

```diff
@@ -269,14 +269,20 @@ public InputStream getInputStreamForSection(FileSummary.Section section,
                                                 String compressionCodec)
         throws IOException {
       FileInputStream fin = new FileInputStream(filename);
-      FileChannel channel = fin.getChannel();
-      channel.position(section.getOffset());
-      InputStream in = new BufferedInputStream(new LimitInputStream(fin,
-          section.getLength()));
+      try {
```

Comment:
We could have:
try (FileInputStream fin = new FileInputStream(filename)) {

}

Reply:
I don't think we can use try-with-resources here, since in the normal case, `in` gets returned, and it is a wrapper for `fin`.

Reply:
[This](https://github.com/apache/hadoop/blob/06a5d3437f68546207f18d23fe527895920c756a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSImageFormatPBINode.java#L236) is a call site invoking this method where InputStream gets closed by the caller.

Reply:
That's true we want to keep it open.

Reply:
Thanks a lot for the review. I really appreciate it. 
I was wondering if I need to do anything else to get the change merged.